server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -692,8 +692,13 @@ public enum FieldExtractPreference {
         /**
          * No preference. Leave the choice of where to load the field from up to the FieldType.
          */
-        NONE;
-
+        NONE,
+        /**
+         * Prefer loading from stored fields like {@code _source} because we're
+         * loading many fields. The {@link MappedFieldType} can chose a different
+         * method to load the field if it needs to.
+         */
+        PREFER_STORED;
```

Comment:
I feel like the prefix `PREFER_` duplicates the enum name. None of the other enum values have `PREFER_` as a prefix, so perhaps this should not. The fact that this is a _preference_ is already in the `FieldExtractPreference`.

Reply:
Sure. I feel like the other "preference"s were more like requirements. But I'll rename.

Reply:
The others were intended to be preferences. The previous default behaviour of loading doc-values if they exist, otherwise source, was swapped around for spatial types, and the `DOC_VALUES` preference was to swap that back, for performance reasons, but we could still load from source (just slower). Also the bounds extraction preference is just a performance optimization, because if we don't do that we still get the right answer, just much slower.